### PR TITLE
Adjust to the changes of `yiisoft/rbac` package 

### PR DESCRIPTION
**Pull request by razonyang**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

## Summaries

* Fix primary keys of assignments table
* Implement the `exists` method of `ItemsStorageInterface` for `ItemsStorage`.
* Bump yiisoft/rbac to `^1.0`.
* Fix `ItemsStorage::getChildren` to follow the return type `array<string,Item>`.
* Fix the compatibility and test case for MySQL and PostgreSQL: replace `select('1')` with `select([new Fragment('1')])`  to avoid the *unknown column '1' in the 'field list'* error.

### RbacCycleInit command

The `RbacCycleInit` command failed to generate the assignments table,

```sql
SQLSTATE[42000]: Syntax error or access violation: 1063 Incorrect column specifier for column 'itemName'

Creating `auth_assignment` table...
 > SELECT COUNT(*) FROM `information_schema`.`tables` WHERE `table_schema` = 'newtab' AND `table_name` = 'auth_assignment'
 ! CREATE TABLE `auth_assignment` (
    `itemName` varchar (128) NOT NULL AUTO_INCREMENT,
    `userId` varchar (128) NOT NULL AUTO_INCREMENT,
    `createdAt` int (11) NOT NULL,
    CONSTRAINT `auth_assignment_foreign_itemname_630d84f623952` FOREIGN KEY (`itemName`) REFERENCES `auth_item` (`name`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE InnoDB
 ! SQLSTATE[42000]: Syntax error or access violation: 1063 Incorrect column specifier for column 'itemName'
```

The patch's SQL as follows.

```sql
Creating `auth_assignment` table...
 > SELECT COUNT(*) FROM `information_schema`.`tables` WHERE `table_schema` = 'newtab' AND `table_name` = 'auth_assignment'
 > CREATE TABLE `auth_assignment` (
    `itemName` varchar (128) NOT NULL,
    `userId` varchar (128) NOT NULL,
    `createdAt` int (11) NOT NULL,
    PRIMARY KEY (`itemName`, `userId`),
    CONSTRAINT `auth_assignment_foreign_itemname_630d898cbe79a` FOREIGN KEY (`itemName`) REFERENCES `auth_item` (`name`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE InnoDB
 > CREATE INDEX `auth_assignment_index_itemname_userid_630d898cbe793` ON `auth_assignment` (`itemName`, `userId`)
 > CREATE INDEX `auth_assignment_index_itemname_630d898cbe7a2` ON `auth_assignment` (`itemName`)
Table `auth_assignment` created successfully
```

### CODE REVIEW

File: src/ItemsStorage.php [hunk 1]
```diff
@@ -70,6 +71,16 @@ public function get(string $name): ?Item
         return empty($item) ? null : $this->populateItem($item);
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function exists(string $name): bool
+    {
+        $item = $this->database->select([new Fragment('1')])->from($this->tableName)->where(['name' => $name])->run()->fetch();
```
Comment (roxblnfk) [inline src/ItemsStorage.php:79]: I think you can use `count()` here.
But the better case may be adding for `limit(1)` in all similar queries.

File: src/ItemsStorage.php [hunk 2]
```diff
@@ -70,6 +71,16 @@ public function get(string $name): ?Item
         return empty($item) ? null : $this->populateItem($item);
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function exists(string $name): bool
```
Comment (samdark) [inline src/ItemsStorage.php:77]:
```suggestion
    public function exists(string $name): bool
```

**Review by roxblnfk — APPROVED** [review]

**Review by samdark — APPROVED** [review]

File: src/ItemsStorage.php [hunk 3]
```diff
@@ -73,9 +73,16 @@ public function get(string $name): ?Item
...
+            ->database
+            ->select([new Fragment('1')])
+            ->from($this->tableName)
+            ->where(['name' => $name])
+            ->limit(1)
+            ->run()
+            ->fetch();
 
-        return !empty($item);
+        return $result !== false;
```
Comment (roxblnfk) [inline src/ItemsStorage.php:85]: Are you sure the result can be `false`?

**Comment by samdark** [comment]
Thanks!

**Pull request merged**